src/Button.js

```diff
@@ -7,19 +7,59 @@ import { rem, transparentize } from 'polished';
 const verticalAlignSmall = '-4px';
 const verticalAlignSmaller = '-6px';
 
-const sizeStyles = ({ theme, size, text }) => {
+const calculatePadding = ({ xDefault, iconStart, iconEnd, theme, text }) => {
+  if (text === undefined || (iconStart && iconEnd)) {
+    return theme.space.smallest;
+  }
+
+  if (iconStart) {
+    return `${theme.space.smallest} ${theme.space.smaller} ${
+      theme.space.smallest
+    } ${theme.space.smallest}`;
+  }
+
+  if (iconEnd) {
+    return `${theme.space.smallest} ${theme.space.smallest} ${
+      theme.space.smallest
+    } ${theme.space.smaller}`;
+  }
+
+  return `${theme.space.smallest} ${xDefault}`;
+};
+
+const sizeStyles = ({ theme, iconStart, iconEnd, size, text }) => {
   const sizeMap = {
     small: {
-      padding: `0 ${theme.space.smaller}`,
+      padding: (() => {
+        if (text === undefined || (iconEnd && iconStart)) {
+          return 0;
+        }
+
+        if (iconStart) {
+          return `0 ${theme.space.smaller} 0 0`;
+        }
+
+        if (iconEnd) {
+          return `0 0 0 ${theme.space.smaller}`;
+        }
+
+        return `0 ${theme.space.smaller}`;
+      })(),
       height: rem('24px'),
       fontSize: theme.fontSizes.size1,
       i: {
         fontSize: rem('24px'),
-        verticalAlign: text === undefined ? 0 : verticalAlignSmall
+        verticalAlign: text === undefined ? 0 : '-5px'
```

Comment:
Is there enough difference between the verticalAlignSmall at -4px and this at -5px to warrant this change?

Reply:
Yes, when looking at it again in storybook, it felt clearly misaligned.